### PR TITLE
Feat/card4l

### PR DESCRIPTION
**Pull request by wildintellect**
Before you submit a pull request, please fill in the following:

**Related Issue(s):**
Please add links (if any) to related github issues.
#3 , Please merge #2 first, this branch is based on that branch.

**Description:**
First pass at getting the ScanSAR product STAC records generated. This includes reading item values from Card4l xml metadata files.

**PR checklist:**

- [x] Code is formatted (run `scripts/format`).
- [x] Code lints properly (run `scripts/lint`).
- [x] Tests pass (run `scripts/test`).
- [x] Documentation has been updated to reflect changes, if applicable.
- [x] Examples have been updated to reflect changes, if applicable
- [x] Changes are added to the [CHANGELOG](../CHANGELOG.md).

### CODE REVIEW

File: examples/palsar2-scansar.json [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    {
+      "rel": "self",
+      "href": "/workspaces/palsar2-scansar/examples/palsar2-scansar.json",
+      "type": "application/json"
+    }
+  ],
+  "stac_extensions": [
+    "https://stac-extensions.github.io/item-assets/v1.0.0/schema.json",
+    "https://stac-extensions.github.io/projection/v1.0.0/schema.json",
+    "https://stac-extensions.github.io/raster/v1.1.0/schema.json",
```
Comment (jjfrench) [inline examples/palsar2-scansar.json]: Is the raster extension only included for the DataType in constants? Or is there hope that they'll fill in raster:bands?

Comment (wildintellect) [reply examples/palsar2-scansar.json]: Should be fixed now.

File: examples/collection.json [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+      "title": "The new PALSAR-2 ScanSAR products have been developed to conform with the CEOS-ARD format specifications for Normalised Radar Backscatter (CARD4L NRB). The NRB products have been subject to ortho-rectification and Radiometric Terrain Correction (RTC), and are provided in the Gamma-0 backscatter convention, which mitigates the variations from diverse observation geometries. The products include HH and HV polarisation backscatter, local incidence angle image and data mask.",
+      "description": "Also includes data usage information"
+    },
+    {
+      "rel": "license",
+      "href": "https://earth.jaxa.jp/policy/en.html",
+      "type": "JAXA Terms of Use of Research Data"
+    },
+    {
+      "rel": "self",
```
Comment (jjfrench) [inline examples/collection.json]: - in the process of removing "self" and normalizing hrefs

Comment (jjfrench) [reply examples/collection.json]: @wildintellect I think that we want a create-examples.py for these, similar to https://github.com/stactools-packages/sentinel1/blob/main/scripts/create_examples.py

File: src/stactools/palsar2_scansar/constants.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        roles=[PR.PRODUCER, PR.PROCESSOR, PR.LICENSOR],
+        url="https://www.eorc.jaxa.jp/ALOS/en/dataset/fnf_e.htm",
+    ),
+    Provider(
+        "Japan Aerospace Exploration Agency",
+        roles=[PR.HOST],
+        url="https://registry.opendata.aws/jaxa-alos-palsar2-scansar/",
+    ),
+]
+SCANSAR_DESCRIPTION = """The new PALSAR-2 ScanSAR products have been developed to conform with the CEOS-ARD format specifications for Normalised Radar Backscatter (CARD4L NRB). The NRB products have been subject to ortho-rectification and Radiometric Terrain Correction (RTC), and are provided in the Gamma-0 backscatter convention, which mitigates the variations from diverse observation geometries. The products include HH and HV polarisation backscatter, local incidence angle image and data mask."""  # noqa: E501
```
Comment (jjfrench) [inline src/stactools/palsar2_scansar/constants.py:35]: The life-long debate of "polarisation" vs "polarization". I would say that if we're using the SAR extension to use the "Z" but I'm open to opinions

File: src/stactools/palsar2_scansar/constants.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+SCANSAR_REVISION = "E"
+
+SCANSAR_LINKS = [
+    Link(
+        "documentation",
+        ("https://www.eorc.jaxa.jp/ALOS/en/dataset/palsar2_l22_e.htm"),
+        "application/pdf",
+        SCANSAR_DESCRIPTION,
```
Comment (jjfrench) [inline src/stactools/palsar2_scansar/constants.py]: The title for the documentation link is the entire SCANSAR description. Should we slim it down?

File: src/stactools/palsar2_scansar/constants.py (hunk 3 above)
Comment (wildintellect) [reply src/stactools/palsar2_scansar/constants.py:35]: I copy/pasted that from the official documentation by JAXA. Since it's not a variable name, I think no change.

File: src/stactools/palsar2_scansar/constants.py [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+            "title": "summary",
+            "type": pystac.MediaType.XML,
+            "description": "CARD4L Metadata",
+            "role": "metadata",
+        }
+    ),
+    "kml": AssetDefinition(
+        {
+            "title": "kml",
+            "type": pystac.MediaType.XML,
```
Comment (jjfrench) [inline src/stactools/palsar2_scansar/constants.py]: Double-checking, is this the preferred content-type for KML?

File: src/stactools/palsar2_scansar/constants.py [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+            "description": "",
```
Comment (jjfrench) [inline src/stactools/palsar2_scansar/constants.py]: Not sure what should go here for the description

File: src/stactools/palsar2_scansar/constants.py (hunk 5 above)
Comment (wildintellect) [reply src/stactools/palsar2_scansar/constants.py]: Good catch, updated to IANA type for KML `application/vnd.google-earth.kml+xml`

**Review by gadomski — COMMENTED** [review]
Can you do me a favor and merge **main** into this branch? I don't seem to have permissions to do so, and I'm currently seeing a lot of changes from #2 in this PR review pane. Thanks!

**Comment by wildintellect** [comment]
@gadomski interesting, I had already merged all those changes before the PR #2 went through. Would it be cleaner to reopen a new PR?
```
$ git merge main
Already up to date.
```

**Comment by wildintellect** [comment]
Wait spoke too soon, let me cross check the forks.

File: .mdlrc [hunk 7]
```diff
@@ -0,0 +1 @@
+## markdownlint config
```
Comment (gadomski) [inline .mdlrc]: Is this file needed?

File: examples/ALOS2437590500-220630_WWDR2.2GUA/ALOS2437590500-220630_WWDR2.2GUA.json [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+  "stac_version": "1.0.0",
+  "id": "ALOS2437590500-220630_WWDR2.2GUA",
+  "properties": {
+    "proj:epsg": 32651,
+    "proj:bbox": [
+      374612.5,
+      2689112.5,
+      780462.5,
+      3087012.5
+    ],
```
Comment (gadomski) [inline examples/ALOS2437590500-220630_WWDR2.2GUA/ALOS2437590500-220630_WWDR2.2GUA.json:12]: This is more personal preference than a rule, but I don't find `proj:bbox` very useful ... do you have a use-case for it?

File: examples/ALOS2437590500-220630_WWDR2.2GUA/ALOS2437590500-220630_WWDR2.2GUA.json [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+    {
+      "rel": "parent",
+      "href": "../collection.json",
+      "type": "application/json",
+      "title": "PALSAR-2 ScanSAR NRB L2.2"
+    }
+  ],
+  "assets": {
+    "HH_SLP": {
+      "href": "ALOS2437590500-220630_WWDR2.2GUA_HH_SLP.tif",
```
Comment (gadomski) [inline examples/ALOS2437590500-220630_WWDR2.2GUA/ALOS2437590500-220630_WWDR2.2GUA.json]: Does this file live in `tests/data-files`? If so, it'd be good to fix up this href to point to the file.

File: src/stactools/palsar2_scansar/stac.py [hunk 10]
```diff
@@ -30,88 +38,147 @@ def create_collection() -> Collection:
...
-        bbox=[-180, 90, 180, -90],
-        datetime=demo_time,
-        stac_extensions=[],
-    )
+        # It is a good idea to include proj attributes to optimize for libs like stac-vrt
+        proj_attrs = ProjectionExtension.ext(item, add_if_missing=True)
+        proj_attrs.epsg = dataset.crs.to_epsg()
+        proj_attrs.bbox = bbox
+        proj_attrs.shape = product_metadata.get_shape  # Raster shape ProductImageSize
+        # proj_attrs.transform = [-180, 360, 0, 90, 0, 180]  # Raster GeoTransform
```
Comment (gadomski) [inline src/stactools/palsar2_scansar/stac.py]: Transform would be good to include, as it's used by (e.g.) **odc-stac** and **stackstac** to load data.

**Review by gadomski — CHANGES_REQUESTED** [review]
Thanks for the branch update, that was much nicer :-)

File: .mdlrc (hunk 7 above)
Comment (wildintellect) [reply .mdlrc]: ah I have a manual override in the README.md because I couldn't get this file to work yet.

File: examples/ALOS2437590500-220630_WWDR2.2GUA/ALOS2437590500-220630_WWDR2.2GUA.json (hunk 8 above)
Comment (wildintellect) [reply examples/ALOS2437590500-220630_WWDR2.2GUA/ALOS2437590500-220630_WWDR2.2GUA.json:12]: I was following https://github.com/stac-extensions/projection#best-practices , not sure if I've seen real world use of it, but reading that doc I can see some potential uses like building a VRT from the STAC records.

Comment (gadomski) [reply examples/ALOS2437590500-220630_WWDR2.2GUA/ALOS2437590500-220630_WWDR2.2GUA.json:12]: Ah gotcha -- yeah I guess I was more used to seeing `transform` used instead of `bbox`, but it's no problem to include. Thanks!

File: src/stactools/palsar2_scansar/stac.py (hunk 10 above)
Comment (wildintellect) [reply src/stactools/palsar2_scansar/stac.py]: oops, just forgot to go back and fix that.

File: examples/ALOS2437590500-220630_WWDR2.2GUA/ALOS2437590500-220630_WWDR2.2GUA.json (hunk 9 above)
Comment (wildintellect) [reply examples/ALOS2437590500-220630_WWDR2.2GUA/ALOS2437590500-220630_WWDR2.2GUA.json]: Ah, for some reason I thought I was supposed to drop that. Fixed.

File: .mdlrc (hunk 7 above)
Comment (wildintellect) [reply .mdlrc]: ok, figured it out, `markdownlint-cli` has a different config than `markdownlint`. This doc was critical to figuring it out. https://lyz-code.github.io/blue-book/devops/markdownlint/

**Review by gadomski — APPROVED** [review]

**Review by jjfrench — APPROVED** [review]

**Comment by jjfrench** [comment]
I missed this in the merge changes but we may need to update the README at some point to 
- include the other [extensions](https://github.com/developmentseed/palsar2-scansar/blob/feat/card4l/README.md?plain=1#L10) used
- update the [description/usage](https://github.com/developmentseed/palsar2-scansar/blob/feat/card4l/README.md?plain=1#L14)
- update the [item example link](https://github.com/developmentseed/palsar2-scansar/blob/feat/card4l/README.md?plain=1#L19)
